### PR TITLE
Hoist Model Refactor

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

Review and check off the below. Items that do not apply can also be checked off to indicate they
have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or open as a draft PR / add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Reviewed for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: core/HoistModel.js [hunk 1]
```diff
@@ -27,7 +28,29 @@ import {applyMixin} from '@xh/hoist/utils/js';
...
-        includes: [ManagedSupport, ReactiveSupport, XhIdSupport]
+        includes: [ManagedSupport, ReactiveSupport, XhIdSupport],
+
+        provides: {
+            lookupModel(selector) {
+                if (this.matchesSelector(selector)) return this;
+
+                let ret = null;
+                forOwn(this, (value, key) => {
+                    if (key.endsWith('Model') && value.isHoistModel && value.matchesSelector(selector)) {
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply core/HoistModel.js]: interesting!  I would say no, and yes!

File: core/hooks/UseModel.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import {useState, useContext} from 'react';
+import {ModelLookupContext} from '../impl/ModelLookup';
+import {useOnUnmount} from '@xh/hoist/utils/react';
+
+/**
+ * Hook to allow a component to access a HoistModel provided in context by an ancestor component.
+ *
+ * @param {(Class|string)} [selector] - class or name of mixin applied to class of
+ *      model to be returned.  If not provided the 'closest' inherited model will be returned.
+ * @param {Array} [props] - provide to do initial search in props.
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply core/hooks/UseModel.js]: That just cruft at this point from an earlier implementation.  Will whack it

File: admin/tabs/activity/tracking/ActivityDetail.js [hunk 3]
```diff
@@ -25,44 +23,48 @@ export const activityDetail = hoistElemFactory(
...
-        )
-    ];
-}
\ No newline at end of file
+const detail = hoistCmpFactory(
+    ({model}) => {
+        const rec = model.detailRecord,
+            user = rec.impersonating ? `${rec.username} as ${rec.impersonating}` : rec.username;
+
+        return [
```
Comment (amcclain) [inline admin/tabs/activity/tracking/ActivityDetail.js]: I found that I was getting react console warnings here re. the need for stable keys when returning the collection directly. Believe we want to wrap in a fragment or a div.

Comment (lbwexler) [reply admin/tabs/activity/tracking/ActivityDetail.js]: Good catch -- had just been reading more about fragments yesterday and was reminded of that benefit w.r.t to key warnings.  Will be sure to respect their power more going forward!

**Pull request merged**